src/tasklistClass.js

```diff
@@ -0,0 +1,118 @@
+class Tasks {
+  constructor() {
+    this.tasks = [];
+    this.items = 0;
+    this.idGen = 0;
+  }
+
+  #addToLocalStorage = () => {
+    localStorage.setItem('tasks', JSON.stringify(this.tasks));
+  }
+
+  #getFromLocalStorage = () => {
+    const storedTasks = JSON.parse(localStorage.getItem('tasks'));
+    if (storedTasks) {
+      this.tasks = storedTasks;
+      return true;
+    }
+    return false;
+  }
+
+  #addToDom = (element, taskListDom) => {
+    taskListDom.appendChild(element);
+  }
+
+  #addEvents = (id, li) => {
+    const deleteButton = li.querySelector(`#delete-${id}`);
+    const inputText = li.querySelector(`#input-${id}`);
+
+    li.addEventListener('dragstart', (e) => {
+      e.dataTransfer.setData('text/plain', e.target.id);
+    });
+    li.addEventListener('dragover', (e) => {
+      e.preventDefault();
+    });
+    li.addEventListener('drop', (e) => {
+      e.preventDefault();
+      const data = e.dataTransfer.getData('text/plain');
+      const draggableElement = document.getElementById(data);
+      const dropzone = e.target.closest('li');
+      dropzone.parentNode.insertBefore(draggableElement, dropzone);
+    });
+
+    inputText.addEventListener('change', (e) => {
+      this.edit(e.target.value, id);
+    });
+
+    li.addEventListener('click', () => {
+      li.classList.add('highlight');
+      li.querySelector('.fa-ellipsis-vertical').classList.add('hidden');
+      li.querySelector('.fa-trash-can').classList.remove('hidden');
+    });
+
+    deleteButton.addEventListener('click', () => {
+      this.remove(id);
+      li.remove();
+    });
+
+    document.addEventListener('click', (e) => {
+      if (!li.contains(e.target)) {
+        li.classList.remove('highlight');
+        li.querySelector('.fa-ellipsis-vertical').classList.remove('hidden');
+        li.querySelector('.fa-trash-can').classList.add('hidden');
+      }
+    });
+  }
+
+  #TaskElement = (task) => {
+    const li = document.createElement('li');
+    li.classList = 'p-4 task-list-element list-group-item d-flex align-items-center justify-content-between';
+    li.id = `task-${task.uniqueId}`;
+    li.setAttribute('draggable', true);
+    li.innerHTML = `<div class="d-flex w-100"><input class="form-check-input me-2" type="checkbox" value=""><input class="w-100 p-0 m-0 border-0" name="" id="input-${task.uniqueId}" value="${task.description}"></div><div><a id="order-${task.uniqueId}"><i class="ps-2 fa-solid fa-ellipsis-vertical"></i></a><a id="delete-${task.uniqueId}"><i class="ps-2 hidden fa-solid fa-trash-can"></i></a></div>`;
+    return li;
+  }
+
+  generate = (taskListDom) => {
+    let indexUpdater = 0;
+    taskListDom.innerHTML = '';
+    this.tasks.forEach((task) => {
+      indexUpdater += 1;
+      task.index = indexUpdater;
+      const li = this.#TaskElement(task);
+      this.#addEvents(task.uniqueId, li);
+      this.#addToDom(li, taskListDom);
+      this.#addToLocalStorage();
+    });
+  }
```

Comment:
- According to the project requirements, Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique. Please fix this 🙏 

Your project:
<img width="453" alt="vwvrvwrv" src="https://user-images.githubusercontent.com/73607512/213700446-9c067e13-87b3-4c35-81b4-fc892655571a.PNG">